include/aws/io/tls_channel_handler.h

```diff
@@ -191,6 +191,19 @@ struct aws_tls_ctx_options {
      * implementation.
      */
     void *ctx_options_extension;
+
+    /**
+     * Set if using PKCS#11 for private key operations.
+     * See aws_tls_ctx_pkcs11_options for more details.
+     */
+    struct {
+        struct aws_pkcs11_lib *lib;                  /* required */
+        struct aws_string *user_pin;                 /* optional */
```

Comment:
user_pin is optional ? Is it not required for sign/decrypt operations?

Reply:
it's NULL if the token uses a "protected authentication path" (token itself has a PIN pad)

I clarified the comment